File: C4_InnerTube/InnerTubeCS/SharedUtilities/ProcessHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Diagnostics;
using System.IO;

namespace SharedUtilities
{
    public enum ProcessType
    {
        iTunes,
        Zune
    }
    
    public static class ProcessHelper
    {
        private static string iTunesExe = "iTunesHelper";
        private static string ZuneExe = "ZuneLauncher";


        public static bool IsProcessRunning(string name)
        {
            var x = Process.GetProcesses();
            var result = from p in x
                         where p.ProcessName == name
                         select p;

            if (result.Count() > 0)
            {
                return true;
            }
            {
                return false;
            }
        }

        public static bool IsProcessRunning(ProcessType process)
        {
            switch (process)
            {
                case ProcessType.iTunes:
                    return IsProcessRunning(iTunesExe);
                case ProcessType.Zune:
                    return IsProcessRunning(ZuneExe);    
                default:
                    return false;
                    
            }
        }
        
    }
}

File: C4_InnerTube/InnerTubeVB/SharedUtilities/ProcessHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Diagnostics
Imports System.IO

Namespace SharedUtilities
	Public Enum ProcessType
		iTunes
		Zune
	End Enum

	Public NotInheritable Class ProcessHelper
		Private Shared iTunesExe As String = "iTunesHelper"
		Private Shared ZuneExe As String = "ZuneLauncher"

        Public Shared Function IsProcessRunning(ByVal name As String) As Boolean
            Dim x = Process.GetProcesses()
            Dim result = From p In x _
                         Where p.ProcessName = name _
                         Select p

            If result.Count() > 0 Then
                Return True
            End If
            Return False
        End Function

		Public Shared Function IsProcessRunning(ByVal process As ProcessType) As Boolean
			Select Case process
				Case ProcessType.iTunes
					Return IsProcessRunning(iTunesExe)
				Case ProcessType.Zune
					Return IsProcessRunning(ZuneExe)
				Case Else
					Return False

			End Select
		End Function

	End Class
End Namespace
